src/desktop/ipc/CentralIpcHandler.ts

```diff
@@ -16,26 +16,43 @@ type RequestHandler = (message: Message<unknown>) => unknown
  * registered once per event name so this must be a singleton per event name!
  */
 export class CentralIpcHandler<IpcConfigType extends IpcConfig<string, string>> {
-	private readonly webContentsIdToRequestHandler = new Map<number, RequestHandler>()
+	// using ref to webContents because webContents tend to get destroyed
+	// and we don't know if ids get reused.
+	private readonly webContentsToRequestHandler = new Map<WebContents, RequestHandler>()
```

Comment:
can this be a case for `WeakMap`? `WebContents` should be help by a window anyway, right?

Reply:
I think that's a good ideo, then we can just check if the WC is in the map and don't worry about removing anything.
since the only need to iterate the map is to find destroyed WCs, the fact that you can't iterate a WeakMap shouldn't matter